### PR TITLE
Fix false interpreter

### PR DESCRIPTION
**Pull request by bhansconnect**
Gets the false interpreter upgraded to purity inference, cleaned up in general, and running as a test again.

### CODE REVIEW

File: crates/cli/tests/test-projects/false-interpreter/main.roc [hunk 1]
```diff
@@ -11,132 +11,124 @@ import Variable exposing [Variable]
 # It has some extra constraints:
 # 1) The input files are considered too large to just read in at once. Instead it is read via buffer or line.
 # 2) The output is also considered too large to generate in memory. It must be printed as we go via buffer or line.
+
+# All of the notes below this line are hopefully outdated with purity inference.
```
Comment (smores56) [inline crates/cli/tests/test-projects/false-interpreter/main.roc:15]: nitpick: maybe prefix this with a TODO so we remember to remove the outdated notes if this successfully fixes the issue they point out?

**Review by smores56 — APPROVED** [review]
So much cleaner with purity inference. Good change!

File: crates/cli/tests/test-projects/false-interpreter/main.roc (hunk 1 above)
Comment (bhansconnect) [reply crates/cli/tests/test-projects/false-interpreter/main.roc:15]: Good call. Removed. I think it is all resolved.

Comment (bhansconnect) [reply crates/cli/tests/test-projects/false-interpreter/main.roc:15]: Also, false feel surprisingly snappy for what it is.

**Review by smores56 — DISMISSED** [review]

**Comment by bhansconnect** [comment]
Ok, this finally should be good to merge. 🤞 all tests pass.

**Comment by bhansconnect** [comment]
Note, I thought the drop specialization changes where getting merged into main. Actually got merged into this branch/pr. They should all be good.

**Review by smores56 — APPROVED** [review]

**Pull request merged**